tests/test_app/TestApp/Model/Table/TagsTable.php

```diff
@@ -27,4 +29,14 @@ public function initialize(array $config): void
         $this->belongsToMany('Articles');
         $this->hasMany('ArticlesTags', ['propertyName' => 'extraInfo']);
     }
+
+    public function findSlugged(SelectQuery $query): SelectQuery
+    {
+        return $query->applyOptions(['preserveKeys' => true])
+            ->formatResults(function ($results) {
+                return $results->indexBy(function ($record) {
```

Comment:
If this `formatResults -> indexBy` chain becomes commonly used we could add a method for it with preservekeys on by default.